src/lib.rs

```diff
@@ -233,24 +300,55 @@ This report was created using `recrep` for {{organization}}/{{application}}/{{ve
     }
 
     fn crashes_from_api(&self, api: impl API) -> Result<Report, &'static str> {
-        let crash_manager = crashes::CrashManager {};
-        crash_manager.crash_list(
+        let crash_downloader = crashes::CrashManager {};
+
+        crash_downloader.crash_list(
             &api,
             self.organization.as_str(),
             self.application.as_str(),
             self.version.clone(),
             self.distribution_group.clone(),
         )
     }
+
+    fn download_group_details_for_crashes(
+        &self,
+        crash_report: &Report,
+    ) -> HashMap<String, Vec<OperatingSystemCount>> {
+        let api = AppCenter::new(self.token.clone());
+        let crash_downloader = crashes::CrashManager {};
+
+        let mut error_groups: HashMap<String, Vec<OperatingSystemCount>> = HashMap::new();
+        for crash in crash_report.crash_list.crashes.iter() {
+            if let Some(error_group_id) = &crash.error_group_id {
+                match crash_downloader.error_group_details(
+                    &api,
+                    error_group_id.as_str(),
+                    self.application.as_str(),
+                    self.organization.as_str(),
+                ) {
+                    Ok(error_group) => {
+                        error_groups.insert(
+                            crash.error_group_id.as_ref().unwrap().to_string(),
+                            error_group.operating_systems,
+                        );
+                    }
+                    Err(_e) => continue,
```

Comment:
We are not interested in this error message?

Reply:
I added some debug information. Good point.